lib/streamlit/type_util.py

```diff
@@ -281,23 +300,30 @@ def is_dataframe_like(obj: object) -> TypeGuard[DataFrameLike]:
     return any(is_type(obj, t) for t in _DATAFRAME_LIKE_TYPES)
 
 
-def is_snowpark_or_pyspark_data_object(obj: object) -> bool:
-    """True if if obj is of type snowflake.snowpark.dataframe.DataFrame, snowflake.snowpark.table.Table or
-    True when obj is a list which contains snowflake.snowpark.row.Row or True when obj is of type pyspark.sql.dataframe.DataFrame
-    False otherwise.
+def is_unevaluated_data_object(obj: object) -> bool:
+    """True if the object is one of the supported unevaluated data objects.
```

Comment:
The comment says *what* the function is doing but not *why* we are doing that. I would find it helpful if you could add a comment to why they are not supported and perhaps also what the delta to supporting them is. Is it something on our side we have to implement or on the format's side?